### ISSUE TITLE
ko.applyBindingsToNode component binding throws error

### ISSUE THREAD

**Issue by siimnelis** [knockout/knockout#2343]
In beta version 3.5 line 4333 throws error, when using component binding with ko.applyBindingsToNode

```
var element= document.createElement("div");
                ko.applyBindingsToNode(element { component: { name: "example", params:{} } });
```

childBindingContext = asyncContext.createChildContext(componentViewModel, /* dataItemAlias */ undefined, function(ctx) {

asyncContext is undefined

Problem originates from line 3450 

```
var bindingInfo = ko.utils.domData.get(node, boundElementDomDataKey);

where ko.utils.domData.get(node, boundElementDomDataKey); returns undefined
```

Error code from console:

```
knockout-3.5.0beta.debug.js?cache=53027963:4333 Uncaught TypeError: Cannot read property 'createChildContext' of undefined
    at ko.subscription._callback (knockout-3.5.0beta.debug.js?cache=53027963:4333)
    at ko.subscribable.notifySubscribers (knockout-3.5.0beta.debug.js?cache=53027963:1412)
    at knockout-3.5.0beta.debug.js?cache=53027963:3843
    at knockout-3.5.0beta.debug.js?cache=53027963:3861
    at knockout-3.5.0beta.debug.js?cache=53027963:3890
    at init.js?cache=53027963:35
    at tryIssueCallback (knockout-3.5.0beta.debug.js?cache=53027963:4000)
    at knockout-3.5.0beta.debug.js?cache=53027963:4021
    at knockout-3.5.0beta.debug.js?cache=53027963:3890
    at resolveViewModel (knockout-3.5.0beta.debug.js?cache=53027963:4071)
```

**Comment by brianmhunt** [comment]
@siimnelis Thanks for the report. The code being run here isn't immediately apparent. Perhaps you could provide a jsFiddle illustrating the problem?

**Comment by siimnelis** [comment]
Is there an http url I can link to version 3.5.

**Comment by siimnelis** [comment]
Is there somewhere an http url for version 3.5?

**Comment by brianmhunt** [comment]
https://unpkg.com/knockout@3.5.0-beta/build/output/knockout-latest.js

**Comment by siimnelis** [comment]
Thank you. Here is the fiddle. 

https://jsfiddle.net/4eg0w6xm/1/

**Comment by siimnelis** [comment]
Any thoughts?

**Comment by pimbrouwers** [comment]
Something jumps out to me is the `self = {};` assignment. Should be `self = this;`.

**Comment by siimnelis** [comment]
It's not a problem. I'm just using functional inheritance. 

http://julien.richard-foy.fr/blog/2011/10/30/functional-inheritance-vs-prototypal-inheritance/

**Comment by siimnelis** [comment]
Any progress?

**Comment by brianmhunt** [comment]
Thanks for the bump @siimnelis — I had a look but haven't had a chance to root out the issue.

BTW, have you cross-posted to StackOverflow?  You might get some more immediate attention there.

**Comment by mbest** [comment]
Sorry I haven't had a chance to dig into this yet. It will definitely be addressed before 3.5 is released.
